### ISSUE TITLE
Tracking page suggestion

### ISSUE THREAD

**Issue by mikeruss1** [CIDRAM/CIDRAM#209]
it would be helpful if the IP Tracking page showed an abbreviated reason for the block. It would provide a quick and easy way of detecting trends without having to work through the logs.
it would increase the size of the DB which could be an issue, maybe hold them as codes rather than full text. Maybe CS for Cloud service etc. For UAs just the first few characters.
only first reason needs to be shown I think.
maybe show an indicator if subsequent blocks had different first reasons

**Comment by 737simpilot** [comment]
I have thought about this years ago and just never thought to bring it up. The ideas are interesting and sound prudent about abbreviations, but I think that might want to be at a limited scale, i.e. the UA part of the "why" reason. The UA, while can be faked, is an important part of the whole connection which may have been why it was blocked in the first place. I think for purposes of keeping size down, a compression mechanism can be used, and since this is nothing more than text should compress pretty well. I don't know what PHP is able to do in the art of compression algorithms, but I do know that the phpBB forum software can use gzip and bzip2. I suppose PHP is already capable of using the old tried and true zip, and I'm sure that's what's used now in CIDRAM for log archives. Though, I'm not sure without peeking at the code and knowing what I'm looking at. Which algorithm has better compression for text and its load on the CPU I'm not sure of without doing some research. There would have to be some kind of compromise ratio to A: compression and B: CPU load. Then again, maybe it's very minute. For PDO database entries, I'm not entirely sure how SQL compression works or if code from gzip, bzip2 or zip can be directly inserted into the table. I'm not well versed with database stuff. On that subject, I actually have a program somewhere here that can create database queries based on what you need. I have yet to try it myself. I often times had to ask people at the phpBB forum how to do this and that query for specific things. This program could help someone understand how to create a query and then type one from scratch. 

At any rate, yeah, the first reason and subsequent other reasons also have to be dealt with. I guess in the interest of the KISS (Keep It Simple Stupid) principle, the first "why" reason is what should be recorded as that was "why" the block in the first place. And yeah, subsequent blocks with different "whys" could be indicated I guess buy a simple marker of some sorts.


-The following is off topic-

I really wish I knew PHP because I'd do a pull request and contribute beyond IP submissions, suggestions and monetary donations, but I don't. I just know that with what I've edited PHP wise either in CIDRAM, phpBB or WordPress via help, it's like a whole new language encompassing some logic and possible math just like C++. And all it takes is one character screw up and the whole thing comes crashing down. So you really have to know it well and code for a long time to get used to it all. Then of course when there's new PHP versions those versions have other options and may depreciate what was available in older PHP versions which means you have to stay abreast of all that. Yeah, as much as I like to learn code, and signed up for Code Academy, I put it down right now because I want to learn more about Linux in deploying a LAMP stack and PHP updates and what not. I've done it in VMware with CentOS some time ago, but I need to go back and learn it all again and take notes. I'm just on a shared account at the moment, but I want to go VPS one day for better management in the _security department_. Then there a lot of other things that occupy my interests as well, hence the username. Today, I know most of what every single button and switch does in a 737 NG (Next Generation) and how to fly the beast with SIDs (Standard Instrument Departure) and STARS (Standard Terminal Arrival), program the FMC (FMS) and the _whole nine yards to the fortieth power_. LOL. That was just through sheer dedication and years of Sim flying and of course lots of reading. Don't forget lots of home made lattes, too! So with that, the same would be true for learning code like; PHP, Pearl, CSS, HTML, Ruby, React, C++ or C# ad nauseam.  All of which I'd like to master one day as part of the nerd ethos I have. LOL I also know lots of other things I learned in the spy craft realm of things, radio communications, escape and evasion, and lock picking. Sheer - freaking - dedication with lots of reading. I can also make some great spaghetti with meatballs and Italian sausage. :D LOL! 

Anyway, I've gone way off topic here and need to squeeze a squad for ink. _I'm so wonly, very wonly._ :D

**Comment by Maikuolan** [comment]
Currently got something else I'm working on for CIDRAM which is causing me a bit of grief and headache at the moment (nothing announced anywhere yet, since I'm not sure how much of what I'm currently doing will actually be *possible*, in the current scope of the project currently, so I'm not sure how much of it will actually make it to being committed and pushed). But, I'll look into this soon, plus all the other stuff suggested and discussed, as soon as I either finish up or decide I'm sick enough of it that I want a break and to move onto something else. '^.^

**Comment by mikeruss1** [comment]
re PHP - try it. It is fairly straightforward, and there is a generic similarity with C++. I have trouble understanding professional code, it is generally denser than my simpler stuff so I wouldnt recommend adapting other people's code. I have it set up on my desktop so I can test PHP, Mysql, Apache very easily. The first few hours will be tough with lots of checking the PHP manual. But it is very satisfying. Use something like Netbeans to code, you get instant help with instruction format. The compiler usually provides good error indications, sometimes inter-related errors confuse it. The WAMP I use is wampserver.

**Comment by 737simpilot** [comment]
Yeah, I uses XAMPP for myself. I acctully created one of my websites on my laptop for two months in XAMPP and then uploaded the public_html files and database for live. Though, there's some know how to do that with phpBB and WordPress since you have to change certain database settings before going live. Though, I suppose once it's in the live phpmyadmin, you can make the changes there as well.

Right now I'm studding for a possible amateur radio license. I've been in the radio communications hobby since high school going back some 25 years and while I did try to get my novice and technician license back in the day, I failed because my book was on the old question pool. LOL I never tried again. Now-a-days we have smartphones and I installed a pretty cool App that gives you the answers to all 300 possible questions. So far I'm acing each section when I do a quiz on those sections. I also re-quiz myself on the sections I think are harder to remember. I got about 200 more questions to go. The test is actually only 35 questions out of the possible 300, but you have to know it all. There's another band here called GMRS (General Mobile Radio Service. It's in the UHF ~460 MHz band) and I guess they reduced the license fee or will be reducing the license fee to $35 so I may take advantage of that. It covers your whole immediate family so everyone can use the repeater. In each case, both a ham radio and I think a GMRS radio license lasts ten years before you have to renew. One thing I learned was that a ham radio technician license can only be used down to the 10 meter band (~26 MHz) which I never knew before. So if I really want to get on the shortwave frequencies I'll have to get a higher tier license. Right now, and if memory serves, it's Technician, General and Extra.

**Comment by Maikuolan** [comment]
Some *partial* progress: As of now, when using the front-end IP tracking page and opting to check against signature files, the "why reason" for any of the tracked IPs which happen to be blocked will be displayed in the tracking status for that entry. :-)

**Comment by mikeruss1** [comment]
Good stuff, thank you.

**Comment by 737simpilot** [comment]
Was just testing the crap out of everything and I seem to be having issues with the correct auxiliary rule showing as the block reason. I know this is partial progress however. I do have a very complex list of auxiliary rules and one of my rules seems to be acting up and not doing what it should. I need to do some more testing to help figure this out before I submit an issue.

Beyond the correct auxiliary rules not showing up, the regular IP tracking seems to show the correct block reason. I used my VPN and one IP to verify this.

**Comment by Maikuolan** [comment]
Does it show correctly at the IP test page, or only at the IP tracking page? And to confirm: It behaves incorrectly during normal execution (or not)?

**Comment by 737simpilot** [comment]
It's just the why reason under the IP Tracking page that's now become part of the core. I have an auxiliary rule named "Security" and when I test that rule to trigger it, the why reason under the IP tacking says it's due to two other auxiliary rules that shouldn't be showing up at all as they are completely unrelated to the block I triggered. I still need to test this with one rule at a time and combinations of rules. It could very well be the "Ref " rule (as I call it) that's doing this in my large auxiliary.yaml file. This "Ref" rule that I needed help with from you a long time ago seems to partially block when triggered. _I think_ I need to test it more since another rule may be greylisting me. What happens is that when the "Ref" rule is triggered it's like the page loads as a 200, but with incomplete CSS elements or something. Like a partial block. In a nutshell, my "Ref" rule pretty much states that if the referrer isn't from an HTTPS link, a block occurs. Again, it _may_ also be because of a greylisting rule. 

It's ~1 AM (~3 PM your time) here now. When I have more time tomorrow (within nine hours) I'll test this further to see if I can isolate the possible issue. 

I just reread my post above and I guess I could have worded that better. Just referring to the new feature here about the why reason in the IP Tracking page and a false display of the why reason as it pertains to an auxiliary rule.

**Comment by Maikuolan** [comment]
Closing as implemented (has been a few years since I'd last looked at this, and I don't think realistically I'll end up doing any further work on it at this point).
